native/src/utils/calendarRangeUtils.ts

```diff
@@ -0,0 +1,49 @@
+type MarkedDateType = {
+  selected: boolean
+  startingDay?: boolean
+  endingDay?: boolean
+}
+
+export const getMarkedDates = (startDate: string, endDate: string): Record<string, MarkedDateType> => {
```

Comment:
Please use `DateTime` here too. That way, you can for example use its methods directly instead of having to manually parse things.